### PR TITLE
Add Go test

### LINKED ISSUE #90: Enable Go test after the repo goes public

### PR DESCRIPTION
**Pull request by lblackstone**
Fix #90

### CODE REVIEW

File: examples/examples_go_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+	integration.ProgramTest(t, &test)
+}
+
+func getGoBaseOptions(t *testing.T) integration.ProgramTestOptions {
+	base := getBaseOptions(t)
+	baseGo := base.With(integration.ProgramTestOptions{
+		Verbose: true,
+		Dependencies: []string{
+			"github.com/pulumi/pulumi-kubernetes/sdk/v3",
```
Comment (mikhailshilkov) [inline examples/examples_go_test.go:27]: This looks wrong?

File: examples/go.mod [hunk 2]
```diff
@@ -1,10 +1,15 @@
 module github.com/pulumi/pulumi-aws-native/examples
 
-go 1.14
+go 1.16
+
+replace github.com/pulumi/pulumi-aws-native/sdk => ../sdk
```
Comment (mikhailshilkov) [inline examples/go.mod:5]: Interesting! Will this pick up the dev build of the provider? How does this interact with the `Dependencies` option of `ProgramTestOptions` ?

**Review by mikhailshilkov — APPROVED** [review]
LGTM with a question around dependencies

File: examples/examples_go_test.go (hunk 1 above)
Comment (lblackstone) [reply examples/examples_go_test.go:27]: Haha, whoops. Good catch.

File: examples/go.mod (hunk 2 above)
Comment (lblackstone) [reply examples/go.mod]: I chatted with @stack72 about this and apparently was doing this completely wrong. Updated and should hopefully be working now.

**Pull request merged**